lib/AST/Decl.cpp

```diff
@@ -733,10 +733,15 @@ bool Decl::predatesConcurrency() const {
   if (isa<ClangModuleUnit>(getDeclContext()->getModuleScopeContext()))
     return true;
 
+  // Variables declared in top-level code are @_predatesConcurrency
+  if (const VarDecl *var = dyn_cast<VarDecl>(this)) {
+    return getASTContext().LangOpts.EnableExperimentalAsyncTopLevel &&
+           var->isTopLevelGlobal();
```

Comment:
Should we also check for Swift version < 6 here?

Reply:
I think that should get picked up by `contextRequiresStrictConcurrencyChecking`, which is used by `adjustVarTypeForConcurrecy` to strip off the actor from the type.
I'll throw it in just to ensure that nothing leaks past though.